wwwapp/views.py

```diff
@@ -434,33 +434,35 @@ def unregister_from_workshop_view(request):
 def data_for_plan_view(request):
     data = {}
 
-    user_profiles_raw = set(up for up in UserProfile.objects.all() if WorkshopUserProfile.objects.filter(user_profile=up, year=settings.CURRENT_YEAR, status='Z').exists())
+    participant_profiles_raw = UserProfile.objects.filter(workshop_profile__year=settings.CURRENT_YEAR, workshop_profile__status='Z')
 
-    workshops_raw = Workshop.objects.filter(status='Z', type__year=settings.CURRENT_YEAR)
+    lecturer_profiles_raw = set()
     workshop_ids = set()
     workshops = []
-    for workshop in workshops_raw:
-        record_to_add = dict()
-        record_to_add['wid'] = workshop.id
+    for workshop in Workshop.objects.filter(status='Z', type__year=settings.CURRENT_YEAR):
+        workshop_data = {'wid': workshop.id, 'name': workshop.title,
+                         'lecturers': [lect.id for lect in
+                                       workshop.lecturer.all()]}
+        for lecturer in workshop.lecturer.all():
+            if lecturer not in participant_profiles_raw:
```

Comment:
This is a tricky case - if somebody is a lecturer and participant at the same time, should he be treated like a participant (like you did here) or like a lecturer (the other way around)? I would argue that the second case makes more sense as it's the 'more important position'.

Actually, this situation should never happen because lecturers don't get a qualification status and they are always implicitly accepted - but we should probably add something to ensure this in another part of the app.

Reply:
The desired use-case is that I will want to give lower priority to workshop preferences of lecturers. In this context being a participant is a higher rank, but I agree that this is tricky.

Reply:
Okay, in that case it makes sense, but we should still ensure this never happens. #71